### ISSUE TITLE
[stdlib] Make `InlineFixedVector` work on `CollectionElement`

### ISSUE THREAD

**Issue by JoeLoser** [modular/max#3478]
As mentioned at https://github.com/modularml/mojo/pull/3445#issuecomment-2338791234, we can make `InlineFixedVector` work on memory-only types such as `CollectionElement`. In the latest nightly tonight, `InlineArray` and `Span` have been relaxed to work on `CollectionElement` instead of `CollectionElementNew`, which should unblock this work.

**Comment by gabrieldemarmiesse** [comment]
Isn't `InlinedFixedVector` deprecated? I know the behavior is slightly different from SBO, but I'm not sure this behavior will be needed a lot once https://github.com/modularml/mojo/pull/2825 lands.

**Comment by JoeLoser** [comment]
> Isn't `InlinedFixedVector` deprecated? I know the behavior is slightly different from small buffer optimization, at least the one I implemented, but I'm not sure this behavior will be needed a lot once #2825 lands.

Yeah, but it's unclear how long it will be before we can land SBO, that's all.

**Comment by gabrieldemarmiesse** [comment]
Fair enough!

**Comment by rd4com** [comment]
Hello, 
`InlineFixedVector` does no longer exist,
(`InlineArray[CollectionElement]` is now the main type for inline storage)

**Comment by soraros** [comment]
CC @JoeLoser.
